cmd/tailscaled/tailscaled.go

```diff
@@ -96,6 +97,54 @@ var subCommands = map[string]*func([]string) error{
 	"debug":                   &debugModeFunc,
 }
 
+// Tailscale 1.14 and before stored state under %LocalAppData%
+// (usually "C:\WINDOWS\system32\config\systemprofile\AppData\Local"
+// when tailscaled.exe is running as a non-user system service).
+// However it is frequently cleared for almost any reason: Windows
+// updates, System Restore, even various System Cleaner utilities.
+//
+// We carefully move any state in AppData\Local to ProgramData
+// at startup. If moving the state fails, we continue to use
+// AppData\Local so at least there will be connectivity.
+func tryWindowsAppDataMigration() {
+	if runtime.GOOS != "windows" {
+		return
+	}
+
+	oldDir := filepath.Join(os.Getenv("LocalAppData"), "Tailscale")
+	_, err := os.Stat(oldDir)
+	if os.IsNotExist(err) {
+		// Common case.
+		return
+	}
+	if err != nil {
+		log.Printf("tryWindowsAppDataMigration failed; AppDataLocal: %v", err)
+		return
+	}
+
+	programData := filepath.Join(os.Getenv("ProgramData"), "Tailscale")
+	_, err = os.Stat(programData)
+	if err == nil {
+		// Both AppData\Local\Tailscale and ProgramData\Tailscale exist. There are
+		// cases where this can happen, like if one downgrades from a version which
+		// stores state in ProgramData back to an earlier version that re-creates
+		// state in AppData\Local.
+		// This is unusual so we log it, but don't otherwise mess with the files.
+		log.Printf("tryWindowsAppDataMigration ProgramData and AppDataLocal both exist")
+		return
+	} else if !os.IsNotExist(err) {
+		log.Printf("tryWindowsAppDataMigration failed; ProgramData: %v", err)
```

Comment:
"... failed: %v" is probably enough, as the error will include ProgramData in it. or replace ProgramData with "new dir: " maybe?

Reply:
Replaced with new dir